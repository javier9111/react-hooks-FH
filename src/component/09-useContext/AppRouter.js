import React from "react";

import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";
import { NavBar } from "./NavBar";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Switch>
            {/* exact sirve para redirigir exactament como lo indica el path asi se evita que la expresion regular tome todo en lugar de lo que solo se le presente ej. "/" /login */}
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/about" component={AboutScreen} />
            <Route exact path="/login" component={LoginScreen} />

            {/*Este es para crear un default donde las rutas que no esten definidas van a este lugar */}
            <Route component={HomeScreen} />

            {/* Este es por si quieres redireccionar a un gar */}
          </Switch>
        </div>
      </div>
    </Router>
  );
};
