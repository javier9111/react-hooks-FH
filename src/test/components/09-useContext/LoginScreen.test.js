import React from "react";
import { mount } from "enzyme";
import { LoginScreen } from "../../../component/09-useContext/LoginScreen";
import { UserContext } from "../../../component/09-useContext/UserContext";
describe("test on <LoginScreen/>", () => {
  const setUser = jest.fn();
  //this should be call with mount but since I have problem with compability I had to ommit it.
  const wrapper = mount(
    <UserContext.Provider
      value={{
        setUser,
      }}
    >
      <LoginScreen />
    </UserContext.Provider>
  );

  test("should show properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // this test has to be done with mount.
  test("should execute the setUser methoid with the expected argument", () => {
    wrapper.find("button").prop("onClick")();
    //expect(setUser).toHaveBeenCalled();
    expect(setUser).toHaveBeenCalledWith({ id: 123, name: "javier" });
  });
});
