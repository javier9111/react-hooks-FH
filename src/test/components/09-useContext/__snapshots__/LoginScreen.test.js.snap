// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <LoginScreen/> should show properly 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
